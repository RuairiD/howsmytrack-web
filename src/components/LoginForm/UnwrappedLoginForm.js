import React from "react";

import { Button, Icon, Input, Form } from "antd";
import { A } from "lemon-reset";
import apiRoot from "../../apiRoot";

const UnwrappedLoginForm = ({ onSubmit, form, submitted }) => (
    <Form onSubmit={onSubmit} className="hmt-form">
        <Form.Item>
            {form.getFieldDecorator("username", {
                rules: [{ required: true, message: "Please enter your username." }],
            })(
                <Input
                    prefix={<Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />}
                    placeholder="Email"
                    autoFocus
                />,
            )}
        </Form.Item>
        <Form.Item>
            {form.getFieldDecorator("password", {
                rules: [{ required: true, message: "Please enter your password." }],
            })(
                <Input
                    prefix={<Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />}
                    type="password"
                    placeholder="Password"
                />,
            )}
        </Form.Item>
        <Form.Item>
            <Button
                block
                type="primary"
                htmlType="submit"
                disabled={submitted}
            >
                Sign In
            </Button>
            <A href={`${apiRoot}/accounts/password_reset`} style={{ float: "right" }}>
                Forgot password?
            </A>
        </Form.Item>
    </Form>
);

export default UnwrappedLoginForm;
