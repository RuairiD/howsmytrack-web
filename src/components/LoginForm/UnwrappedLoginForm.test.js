import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { act } from "react-dom/test-utils";
import UnwrappedLoginForm from "./UnwrappedLoginForm";

jest.mock("../../apiRoot", () => "http://localhost:8000");

const mockValidForm = {
    getFieldDecorator: jest.fn(() => (c) => c),
    validateFieldsAndScroll: jest.fn((afterValidation) => afterValidation(false, {
        username: "username",
        password: "password",
    })),
};

describe("UnwrappedLoginForm", () => {
    it("renders a blank form for logging in on pageload", () => {
        const wrapper = shallow(
            <UnwrappedLoginForm
                form={mockValidForm}
                onSubmit={() => {}}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders an additional registration button if showRegisterButton is provided", () => {
        const wrapper = shallow(
            <UnwrappedLoginForm
                form={mockValidForm}
                onSubmit={() => {}}
                showRegisterButton
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("shows a RegisterModal if the registration button is clicked", () => {
        const wrapper = mount(
            <UnwrappedLoginForm
                form={mockValidForm}
                onSubmit={() => {}}
                showRegisterButton
            />,
        );

        act(() => wrapper.find("Button").get(1).props.onClick());

        wrapper.update();
        expect(wrapper.find("RegisterModal").get(0).props.isVisible).toBeTruthy();

        act(() => wrapper.find("RegisterModal").get(0).props.onCancel());
        wrapper.update();

        expect(wrapper.find("RegisterModal").get(0).props.isVisible).not.toBeTruthy();
    });

    it("renders large buttons if largeButtons is true", () => {
        const wrapper = shallow(
            <UnwrappedLoginForm
                form={mockValidForm}
                onSubmit={() => {}}
                showRegisterButton
                largeButtons
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
