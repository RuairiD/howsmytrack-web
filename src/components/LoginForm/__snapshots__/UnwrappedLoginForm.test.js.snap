// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnwrappedLoginForm renders a blank form for logging in on pageload 1`] = `
<Form
  className="hmt-form"
  colon={true}
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
>
  <FormItem
    hasFeedback={false}
  >
    <Input
      autoFocus={true}
      placeholder="Email"
      prefix={
        <Icon
          style={
            Object {
              "color": "rgba(0,0,0,.25)",
            }
          }
          type="user"
        />
      }
      type="text"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Input
      placeholder="Password"
      prefix={
        <Icon
          style={
            Object {
              "color": "rgba(0,0,0,.25)",
            }
          }
          type="lock"
        />
      }
      type="password"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Button
      block={true}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Sign In
    </Button>
    <A
      className=""
      href="http://localhost:8000/accounts/password_reset"
      style={
        Object {
          "float": "right",
        }
      }
    >
      Forgot password?
    </A>
  </FormItem>
</Form>
`;
